src/utils.js

```diff
@@ -31,3 +34,27 @@ function excludeKeys(obj, keys) {
 function isObject(obj) {
   return typeof obj === 'object' && obj !== null;
 }
+
+function setDimensions(layout, defaultLayout){
+  defaultLayout = defaultLayout || false;
+  var updated = false;
+  if(!defaultLayout){
+    if(window.dimensions.width !== layout.width){
+      updated = true;
+    }
+  }
+  return window.dimensions = {
+    orientation: (layout.width < layout.height ? 'portrait' : 'landscape'),
```

Comment:
Note this can be a false positive on iPad when using split view after this bug was fixed:
https://github.com/facebook/react-native/issues/16152